public/components/overview/home.tsx

```diff
@@ -3,159 +3,200 @@
+      }),
+      getTargetArea: () => HOME_CONTENT_AREAS.SELECTOR,
+    });
+  };
 
-    for (const status of data.status.statuses) {
-      if (status.id.includes(alertsPluginID)) {
-        alertsExists = true;
-      }
-      if (status.id.includes(anomalyPluginID)) {
-        anomalyExists = true;
-      }
-    }
+  const loadDashboardsState = () => {
+    coreRefs.savedObjectsClient
+      ?.find({
+        type: 'dashboard',
+      })
+      .then((response) => {
+        const savedDashboards = response.savedObjects.reduce((acc, savedDashboard) => {
+          const dashboardAttributes = savedDashboard.attributes as {
+            title: string;
+            timeFrom: string;
+            timeTo: string;
+          };
+          const id = savedDashboard.id.toString();
+          acc[id] = {
+            value: id,
+            label: dashboardAttributes.title,
+            startDate: dashboardAttributes.timeFrom,
+            endDate: dashboardAttributes.timeTo,
+          };
+          return acc;
+        }, {} as DashboardSavedObjectsType);
 
-    setAlertsPluginExists(alertsExists);
-    setAnomalyPluginExists(anomalyExists);
-  } catch (error) {
-    console.error('Error checking plugin installation status:', error);
-  }
-};
+        setDashboardsSavedObjects(savedDashboards);
+        const defaultDashboard = getObservabilityDashboardsId();
 
-const navigateToApp = (appId: string, path: string) => {
-  coreRefs?.application!.navigateToApp(appId, {
-    path: `${path}`,
-  });
-};
+        if (defaultDashboard in savedDashboards) {
+          ObsDashboardStateManager.dashboardState$.next({
+            dashboardTitle: savedDashboards[defaultDashboard].label,
+            dashboardId: defaultDashboard,
+            startDate: savedDashboards[defaultDashboard].startDate ?? 'now-7d',
+            endDate: savedDashboards[defaultDashboard].endDate ?? 'now',
+          });
+          ObsDashboardStateManager.isDashboardSelected$.next(true);
+        } else {
+          setObservabilityDashboardsId(null);
+          ObsDashboardStateManager.dashboardState$.next({
+            startDate: '',
+            endDate: '',
+            dashboardTitle: '',
+            dashboardId: '',
+          });
+          ObsDashboardStateManager.isDashboardSelected$.next(false);
+        }
+      })
+      .catch((error) => {
+        console.error('Error fetching dashboards:', error);
+      });
+  };
 
-export type AppAnalyticsCoreDeps = TraceAnalyticsCoreDeps;
+  const flyout = isFlyoutVisible && (
+    <SelectDashboardFlyout
+      closeFlyout={() => setIsFlyoutVisible(false)}
+      dashboardsSavedObjects={dashboardsSavedObjects}
+      reloadPage={loadDashboardsState}
+    />
+  );
 
-interface HomeProps extends RouteComponentProps, AppAnalyticsCoreDeps {
-  parentBreadcrumbs: ChromeBreadcrumb[];
-}
+  useEffect(() => {
+    registerCards();
+    registerDashboard();
+    registerDashboardsControl();
+    loadHomePage();
```

Comment:
Since all of the functions here are asynchronous, could there be any race conditions? Would any of these processes be dependent on any of the others in any way?

Reply:
No each is independent. The registration for each section between the cards / dashboards / controls should be non reliant on each other. Haven't encountered any issues in testing.